packages/sg721/src/lib.rs

```diff
@@ -1,11 +1,50 @@
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::{Addr, Binary, Decimal, Timestamp};
-use cw_ownable::cw_ownable_execute;
+use cosmwasm_std::{Binary, Coin, Timestamp};
+use cw721_base::{
+    msg::{
+        CollectionExtensionMsg, CollectionInfoMsg, ExecuteMsg as Cw721ExecuteMsg,
+        InstantiateMsg as Cw721InstantiateMsg,
+    },
+    DefaultOptionalCollectionExtensionMsg,
+};
+use cw_ownable::Action;
 use cw_utils::Expiration;
 
-#[cw_ownable_execute]
+pub type RoyaltyInfoResponse = cw721_base::msg::RoyaltyInfoResponse;
+pub use cw721_base::state::RoyaltyInfo;
+
 #[cw_serde]
-pub enum ExecuteMsg<T, E> {
+pub enum ExecuteMsg<
+    // NftInfo extension msg for onchain metadata.
+    TNftExtensionMsg,
+    // CollectionInfo extension msg for onchain collection attributes.
+    TCollectionExtensionMsg,
+    // Custom extension msg for custom contract logic. Default implementation is a no-op.
+    TExtensionMsg,
+> {
+    // ---- sg721 specific msgs ----
+    /// Update specific collection info fields
+    #[deprecated = "Please use Cw721UpdateCollectionInfo instead"]
```

Comment:
A lot of allow deprecated everywhere. What would be the condition in which we can remove these tags? Will it ever meet that condition?

Reply:
As noted in PR description:

```md
...
# Backward Compatibility
For backward compatibility and not breaking 3rd parties (like frontend) legacy msgs, structs, etc are kept - but marked as deprecated.
...
```

So this PR can be deployed without breaking frontends. Once frontends (e.g. launchpad, marketplace) are migrated, then all deprecated types can be removed.